dp/generate_prsn.py

```diff
@@ -0,0 +1,278 @@
+
+        for attr in 'project model institute experiment ensemble_member'.split():
+            try:
+                logger.info('{}: {}'.format(attr, getattr(dataset.metadata, attr)))
+            except Exception as e:
+                logger.info('{}: {}: {}'.format(attr, e.__class__.__name__, e))
+        logger.info('dependent_varnames: {}'.format(dataset.dependent_varnames()))
+
+
+def unique_shape(arrays):
+    '''Ensure each array in dict is the same shape'''
+    shapes = {a.shape for a in arrays.values()}
+    return len(shapes) == 1
+
+
+def is_unique_value(values):
+    '''Given a list ensure there is only one unique value'''
+    return np.unique(values).size == 1
+
+
+def pr_freezing_from_units(unit):
+    '''Given a unit determine which temperature describes freezing'''
+    freezing = Q_(0.0, ureg.degC)
+
+    temp_unit = str(ureg.parse_units(unit))
+    if temp_unit != 'degC':
+        freezing = freezing.to(temp_unit)
+
+    return freezing.magnitude
+
+
+def create_prsn_netcdf_from_source(src, dst, to_delete=['original_name, comment']):
+    '''Using a precipiation netCDF as a template copy over the data applicable
+       to prsn
+    '''
+    # Create the dimensions of the file
+    for name, dim in src.dimensions.items():
+        dst.createDimension(name, len(dim) if not dim.isunlimited() else None)
+
+    # Copy the globals
+    unwanted_globals = [
+        'history',
+    ]
+    dst.setncatts({attr:src.getncattr(attr)
+                   for attr in src.ncattrs() if attr not in unwanted_globals})
+
+    # Create the variables in the file
+    for name, var in src.variables.items():
+        dst.createVariable(name, var.dtype, var.dimensions)
+
+        # Copy the variable attributes
+        dst.variables[name].setncatts({attr:var.getncattr(attr) for attr in var.ncattrs()})
+
+        # we will be replacing pr data with prsn data
+        if name == 'pr':
+            continue
+        logger.debug('Copying {}'.format(name))
+        dst.variables[name][:] = src.variables[name][:]
+
+
+    # change pr metadata to prsn equivalents
+    dst.renameVariable('pr', 'prsn')
+    prsn_var = dst.variables['prsn']
+
+    prsn_var.standard_name = 'snowfall_flux'
+    prsn_var.long_name = 'Precipitation as Snow'
+
+    for item in set(to_delete):
+        if hasattr(prsn_var, item):
+            prsn_var.delncattr(item)
+
+
+def custom_filepath(directory, file):
+    '''Check if directory exists and return the filepath'''
+    if not os.path.exists(os.path.dirname(directory)):
+        os.makedirs(os.path.dirname(directory))
+
+    return os.path.join(directory, file)
+
+
+def create_filepath_from_source(source, new_var, outdir):
+    '''Using the source cmor_filename build a new output filepath with new_var
+       and output directory.
+    '''
+    variable, *rest = source.cmor_filename.split('_')
+
+    suffix = ''
+    for var in rest:
+        suffix += '_' + var
```

Comment:
Sorry I didn't spot this before: Why not use `suffix = '_' + '_'.join(vars)`? One line, clearer. Your call.

Reply:
Your way does look much better!